test/images/agnhost/resource-consumer-controller/controller.go

```diff
@@ -214,7 +240,7 @@ func (c *controller) sendConsumeCustomMetric(w http.ResponseWriter, metric strin
 }
 
 func createConsumerURL(suffix string) string {
-	return fmt.Sprintf("http://%s.%s.svc.cluster.local:%d%s", consumerServiceName, consumerServiceNamespace, consumerPort, suffix)
```

Comment:
are other tests depending on this?
should not keep it as default?

Reply:
With proposed change we will have working DNS suffix, because it is taken from real DNS config. So, if "cluster.local" is configured, it will be picked up. That's the trick.

Reply:
great

Reply:
is there a reason not to use `http://%s.%s.svc:%d%s`?

Reply:
> is there a reason not to use `http://%s.%s.svc:%d%s`? that also resolves to the service and doesn't require custom DNS suffix lookup

I was not sure whether it is needed for someone to use full DNS name or not. So, kept the behaviour - full DNS name having correct suffix. For HPA tests just "svc" is enough.
I don't mind to simplify the PR to just removing this ".cluster.local" part of DNS. Should I?

Reply:
No, leave it as full DNS name, otherwise it'll break Windows. :) It doesn't like partially qualified names, and that would break some tests.

Reply:
so... windows doesn't support the kube approach to dns?

Reply:
So, if it is a must to keep old behavior then, I guess, we keep it as is and can move on.

Reply:
WRT Windows, this is one of a long list of things that don't work right in Windows which were deemed "acceptable because no Windows user would expect them to work", even if a Kube user would.  Sigh.